servers/quarkus-server/src/main/java/org/projectnessie/server/authz/CelAccessChecker.java

```diff
@@ -0,0 +1,184 @@
+    LIST_COMMIT_LOG,
+    READ_ENTRIES,
+    ASSIGN_REFERENCE_TO_HASH,
+    COMMIT_CHANGE_AGAINST_REFERENCE,
+    READ_ENTITY_VALUE,
+    UPDATE_ENTITY,
+    DELETE_ENTITY;
+  }
+
+  @Inject
+  public CelAccessChecker(
+      QuarkusNessieAuthorizationConfig config, CompiledAuthorizationRules compiledRules) {
+    this.config = config;
+    this.compiledRules = compiledRules;
+  }
+
+  @Override
+  public void canCreateReference(AccessContext context, NamedRef ref)
+      throws AccessControlException {
+    canPerformOpOnReference(context, ref, AuthorizationRuleType.CREATE_REFERENCE);
+  }
+
+  @Override
+  public void canAssignRefToHash(AccessContext context, NamedRef ref)
+      throws AccessControlException {
+    canPerformOpOnReference(context, ref, AuthorizationRuleType.ASSIGN_REFERENCE_TO_HASH);
+  }
+
+  @Override
+  public void canDeleteReference(AccessContext context, NamedRef ref)
+      throws AccessControlException {
+    canPerformOpOnReference(context, ref, AuthorizationRuleType.DELETE_REFERENCE);
+  }
+
+  @Override
+  public void canReadEntries(AccessContext context, NamedRef ref) throws AccessControlException {
+    canPerformOpOnReference(context, ref, AuthorizationRuleType.READ_ENTRIES);
+  }
+
+  @Override
+  public void canListCommitLog(AccessContext context, NamedRef ref) throws AccessControlException {
+    canPerformOpOnReference(context, ref, AuthorizationRuleType.LIST_COMMIT_LOG);
+  }
+
+  @Override
+  public void canCommitChangeAgainstReference(AccessContext context, NamedRef ref)
+      throws AccessControlException {
+    canPerformOpOnReference(context, ref, AuthorizationRuleType.COMMIT_CHANGE_AGAINST_REFERENCE);
+  }
+
+  @Override
+  public void canReadEntityValue(AccessContext context, NamedRef ref, ContentsKey key)
+      throws AccessControlException {
+    canPerformOpOnPath(context, key, AuthorizationRuleType.READ_ENTITY_VALUE);
+  }
+
+  @Override
+  public void canUpdateEntity(AccessContext context, NamedRef ref, ContentsKey key)
+      throws AccessControlException {
+    canPerformOpOnPath(context, key, AuthorizationRuleType.UPDATE_ENTITY);
+  }
+
+  @Override
+  public void canDeleteEntity(AccessContext context, NamedRef ref, ContentsKey key)
+      throws AccessControlException {
+    canPerformOpOnPath(context, key, AuthorizationRuleType.DELETE_ENTITY);
+  }
+
+  private void canPerformOpOnReference(
+      AccessContext context, NamedRef ref, AuthorizationRuleType type) {
+    if (!config.enabled()) {
+      return;
+    }
+    boolean allowed =
+        compiledRules.getRules().entrySet().stream()
+            .anyMatch(
+                entry -> {
+                  try {
+                    return entry
+                        .getValue()
+                        .execute(
+                            Boolean.class,
+                            ImmutableMap.of(
+                                "ref",
+                                ref.getName(),
+                                "role",
```

Comment:
Think, this should be named `principal` or even `user`?

Reply:
as mentioned in the other comments, I think `role` gives us slightly more flexibility.